### PR TITLE
Custom usermodel

### PR DESCRIPTION
**Pull request by Ropold**
### Backend Changes

- CustomUser Model implementation
- RoomModel & AppUserModel Updates:
   - RoomModel: Category changed to enum.
- RoomController: Added new endpoints:
   - getUserFavorites
   - addRoomToFavorites
   - removeRoomFromFavorites
- Exception Handling:
   - Added handleImageDeletionException for handling image-related errors.
- AppUserRepository: Created a repository for managing AppUser data.
- SecurityConfig: Implemented OAuth2UserService.
- UserController: Users are now saved in AppUserRepository.
- AppUserService: Added methods:
   - getUserById
   - getUserFavorites
   - addRoomToFavorites
   - removeRoomFromFavorites
- RoomService: Added methods:
   - getActiveRooms
   - getRoomsByIds
   - toggleActiveStatus
 
### Tests:
- ExceptionHandlerTest: Adjustments made.
- UserIntegrationTest: 
   - Added testGetMe_withLoggedInUser_expectUsername.
- AppUserServiceTest: Updated old tests and new tests:
   - getUserById_UserExists_ReturnsUser
   - getUserById_UserDoesNotExist_ThrowsException
   - getUserFavorites_ReturnsFavorites
   - addRoomToFavorites_RoomNotInFavorites_AddsRoom
   - addRoomToFavorites_RoomAlreadyInFavorites_DoesNothing
   - removeRoomFromFavorites_RoomInFavorites_RemovesRoom
   - removeRoomFromFavorites_RoomNotInFavorites_DoesNothing
- CloudinaryServiceTest:
   - uploadImage_ValidImage_ReturnsSecureUrl
   - uploadImage_ThrowsIOException_PropagatesException
   - deleteImage_ValidImageUrl_DeletesImage
   - deleteImage_DestroyThrowsIOException_ThrowsImageDeletionException
- RoomServiceTest: Updated old test and added new tests:
   - getActiveRooms
   - toggleActiveStatus
   - getRoomsByIds
   - deleteRoomWithImage
   - deleteRoomWithoutImage
  
### Frontend Changes
- App: Methods for backend interactions:
   - getUserDetails
   - getAppUserFavorites
   - getAllRooms
   - getAllActiveRooms
   - toggleFavorite
   - handleNewRoomSubmit
- AddRoom: Updated to use the new RoomModel.
- Details: Simplified and moved edit & delete logic to MyRooms.
- Favorites (previously Wishlist):
   - Renamed Wishlist to Favorites.
   - favoritesRooms now part of AppUserModel.
   - Logic for managing favorites implemented.
- Home: Refactored to delegate logic to App.
- SearchBar:
   - updated to new RoomModel: Category changed to select 
   - some css adjustments
- MyRooms(New Component):
   - Implements active/offline status management, edit, and delete functionalities.
   - Methods:
      - handleEditToggle
      - handleCategoryChange
      - handleSaveEdit
      - onFileChange
      - handleDelete
      - handleToggleActiveStatus
- Navbar:
   - Added buttons for Favorites and MyRooms.
- Profile: Updated to receive user info via props.
- Utilities: Added getCategoryDisplayName for handling enum display names.
- CSS: Minor adjustments.

### CODE REVIEW

File: backend/src/main/java/ropold/backend/security/UserController.java [hunk 1]
```diff
@@ -1,31 +1,40 @@
...
+            Optional<AppUser> appUserOpt = appUserRepository.findById(user.getName());
+            return appUserOpt.map(AppUser::id).orElse(SecurityContextHolder.getContext().getAuthentication().getName());
+        }
     }
 
     @GetMapping("/me/details")
     public Map<String, Object> getUserDetails(@AuthenticationPrincipal OAuth2User user) {
         if (user == null) {
-           return Map.of("message", "User not authenticated");
+            return Map.of("message", "User not authenticated");
```
Comment (Ropold) [inline backend/src/main/java/ropold/backend/security/UserController.java:36]: u can change it to a class

File: backend/src/main/java/ropold/backend/service/RoomService.java [hunk 2]
```diff
@@ -20,20 +20,34 @@ public List<RoomModel> getAllRooms() {
         return roomRepository.findAll();
     }
 
+    public List<RoomModel> getActiveRooms() {
+        return roomRepository.findAll().stream()
```
Comment (Ropold) [inline backend/src/main/java/ropold/backend/service/RoomService.java:24]: use findAllBY

File: backend/src/main/java/ropold/backend/service/RoomService.java [hunk 3]
```diff
@@ -56,6 +74,27 @@ public RoomModel updateRoomWithPut(String id, RoomModel roomModel) {
...
+                id,
+                room.name(),
+                room.address(),
+                room.category(),
+                room.description(),
+                room.appUserGithubId(),
+                room.appUserUsername(),
+                room.appUserAvatarUrl(),
+                room.appUserGithubUrl(),
+                !room.isActive(),
```
Comment (Ropold) [inline backend/src/main/java/ropold/backend/service/RoomService.java:91]: use lombok with

File: backend/src/test/java/ropold/backend/controller/RoomControllerIntegrationTest.java [hunk 4]
```diff
@@ -41,57 +42,156 @@ class RoomControllerIntegrationTest {
...
+    private AppUserRepository appUserRepository;
+
     @BeforeEach
     void setup() {
         roomRepository.deleteAll();
-        roomModel = new RoomModel("1", "Gürzenich Saal", "Neumarkt 1, 50667 Köln",
-                "Orchester-Saal", "Ein traditionsreicher Saal für Konzerte und Veranstaltungen.",
-                WishlistStatus.ON_WISHLIST, "https://www.test.de/");
-        roomRepository.save(roomModel);
+        appUserRepository.deleteAll();
```
Comment (Ropold) [inline backend/src/test/java/ropold/backend/controller/RoomControllerIntegrationTest.java:51]: maby use lombok @builder

File: backend/src/test/java/ropold/backend/controller/RoomControllerIntegrationTest.java [hunk 5]
```diff
@@ -41,57 +42,156 @@ class RoomControllerIntegrationTest {
...
+                        "appUserGithubId": "123",
+                        "appUserUsername": "Testuser",
+                        "appUserAvatarUrl": "https://avatars-of-test-user.com/",
+                        "appUserGithubUrl": "https://github.com/Testuser",
+                        "isActive": true,
+                        "imageUrl": "https://res.cloudinary.com/Testuser/image/upload/Testuser/testimage1.jpg"
+                    }
+                ]
+                """));
     }
```
Comment (Ropold) [inline backend/src/test/java/ropold/backend/controller/RoomControllerIntegrationTest.java:100]: add test if username = "other" can add or delete ohter favorites

File: frontend/src/components/AddRoom.tsx [hunk 6]
```diff
@@ -18,13 +25,25 @@ export default function AddRoom() {
...
+            name,
+            address,
+            category: category as any, // Wir casten den String zu Category
+            description,
+            appUserGithubId: props.user,
+            appUserUsername: props.userDetails.login,
+            appUserAvatarUrl: props.userDetails.avatar_url,
+            appUserGithubUrl: props.userDetails.html_url,
+            isActive: true,
+            imageUrl: "",
```
Comment (Ropold) [inline frontend/src/components/AddRoom.tsx:41]: is not save. u can add new room with postman with false data

File: frontend/src/components/Favorites.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    user: string;
+    toggleFavorite: (roomId: string) => void;
+};
+
+export default function Favorites(props: Readonly<FavoritesProps>) {
+    const [favoritesRooms, setFavoritesRooms] = useState<RoomModel[]>([]);
+
+    useEffect(() => {
+        axios
+            .get(`/api/practice-hub/favorites/${props.user}`)
```
Comment (Ropold) [inline frontend/src/components/Favorites.tsx:17]: not save ;)

**Review by Flooooooooooorian — APPROVED** [review]

**Pull request merged**